### ISSUE TITLE
Reboot timeout - Installation never finishes

### ISSUE THREAD

**Issue by schlaggi** [hakwerk/labca#43]
Hello,

I get a timeout when I press "restart LabCA" in the browser, after being able to download the root certificate.
The reboot seems to never finish and in the end I have a root and issuer certificate and the certificate for LabCA is installed but the LabCA installation itself is broken - the admin panel shows "Please wait" ending in a timeout or 404, only the public part seems to work. What could be the reason for that?

Installed LabCA a few times on freshly created minimal debian 11 virtual machines. Always the same 🙁

**Comment by hakwerk** [comment]
Please refer to the [Troubleshooting](https://github.com/hakwerk/labca#troubleshooting) section of the README for how to diagnose what the problem is.

**Comment by schlaggi** [comment]
Okay. I did this more detailed now and everything looks nice but "_cd /home/labca/boulder; docker-compose logs labca_" gives me this:

```
labca_1    | 2022/04/23 18:09:41 GET /final
labca_1    | 2022/04/23 18:09:41 GET /
labca_1    | 2022/04/23 18:09:42 Could not determine location: unknown time zone ERROR! On line 140 in commander script
labca_1    | 2022/04/23 18:09:42 http: panic serving 10.77.77.3:57456: runtime error: index out of range [1] with length 1
labca_1    | goroutine 1328 [running]:
labca_1    | net/http.(*conn).serve.func1()
labca_1    |    /usr/local/go/src/net/http/server.go:1802 +0xb9
labca_1    | panic({0x8e1200, 0xc0003d9ab8})
labca_1    |    /usr/local/go/src/runtime/panic.go:1047 +0x266
labca_1    | main._parseStats({0xc000832db0, 0xc0004b2380})
labca_1    |    /go/src/labca/dashboard.go:235 +0xccf
labca_1    | main.CollectDashboardData({0x9c1508, 0xc0004b2380}, 0xc000fd2400)
labca_1    |    /go/src/labca/dashboard.go:374 +0x6be
labca_1    | main.rootHandler({0x9c1508, 0xc0004b2380}, 0xc000fd2400)
labca_1    |    /go/src/labca/main.go:306 +0x85
labca_1    | net/http.HandlerFunc.ServeHTTP(0x897440, {0x9c1508, 0xc0004b2380}, 0x5)
labca_1    |    /usr/local/go/src/net/http/server.go:2047 +0x2f
labca_1    | main.authorized.func1({0x9c1508, 0xc0004b2380}, 0xc000fd2400)
labca_1    |    /go/src/labca/main.go:2359 +0x21a
labca_1    | net/http.HandlerFunc.ServeHTTP(0xc000fd2300, {0x9c1508, 0xc0004b2380}, 0xd1a080)
labca_1    |    /usr/local/go/src/net/http/server.go:2047 +0x2f
labca_1    | github.com/gorilla/mux.(*Router).ServeHTTP(0xc0001de0c0, {0x9c1508, 0xc0004b2380}, 0xc000fd2200)
labca_1    |    /go/pkg/mod/github.com/gorilla/mux@v1.8.0/mux.go:210 +0x1cf
labca_1    | net/http.serverHandler.ServeHTTP({0xc001120f60}, {0x9c1508, 0xc0004b2380}, 0xc000fd2200)
labca_1    |    /usr/local/go/src/net/http/server.go:2879 +0x43b
labca_1    | net/http.(*conn).serve(0xc0004afae0, {0x9c43a0, 0xc0004c43c0})
labca_1    |    /usr/local/go/src/net/http/server.go:1930 +0xb08
labca_1    | created by net/http.(*Server).Serve
labca_1    |    /usr/local/go/src/net/http/server.go:3034 +0x4e8
labca_1    | 2022/04/23 18:09:46 GET /final
labca_1    | 2022/04/23 18:09:46 GET /
labca_1    | 2022/04/23 18:09:47 Could not determine location: unknown time zone ERROR! On line 140 in commander script
labca_1    | 2022/04/23 18:09:47 http: panic serving 10.77.77.3:57480: runtime error: index out of range [1] with length 1
labca_1    | goroutine 1350 [running]:
labca_1    | net/http.(*conn).serve.func1()
labca_1    |    /usr/local/go/src/net/http/server.go:1802 +0xb9
labca_1    | panic({0x8e1200, 0xc0009f60c0})
labca_1    |    /usr/local/go/src/runtime/panic.go:1047 +0x266
labca_1    | main._parseStats({0xc000833170, 0xc0004b2460})
labca_1    |    /go/src/labca/dashboard.go:235 +0xccf
labca_1    | main.CollectDashboardData({0x9c1508, 0xc0004b2460}, 0xc000fd2b00)
labca_1    |    /go/src/labca/dashboard.go:374 +0x6be
labca_1    | main.rootHandler({0x9c1508, 0xc0004b2460}, 0xc000fd2b00)
labca_1    |    /go/src/labca/main.go:306 +0x85
labca_1    | net/http.HandlerFunc.ServeHTTP(0x897440, {0x9c1508, 0xc0004b2460}, 0x5)
labca_1    |    /usr/local/go/src/net/http/server.go:2047 +0x2f
labca_1    | main.authorized.func1({0x9c1508, 0xc0004b2460}, 0xc000fd2b00)
labca_1    |    /go/src/labca/main.go:2359 +0x21a
labca_1    | net/http.HandlerFunc.ServeHTTP(0xc000fd2900, {0x9c1508, 0xc0004b2460}, 0xd1a080)
labca_1    |    /usr/local/go/src/net/http/server.go:2047 +0x2f
labca_1    | github.com/gorilla/mux.(*Router).ServeHTTP(0xc0001de0c0, {0x9c1508, 0xc0004b2460}, 0xc000fd2800)
labca_1    |    /go/pkg/mod/github.com/gorilla/mux@v1.8.0/mux.go:210 +0x1cf
labca_1    | net/http.serverHandler.ServeHTTP({0xc000cbb3b0}, {0x9c1508, 0xc0004b2460}, 0xc000fd2800)
labca_1    |    /usr/local/go/src/net/http/server.go:2879 +0x43b
labca_1    | net/http.(*conn).serve(0xc0004bda40, {0x9c43a0, 0xc0004c43c0})
labca_1    |    /usr/local/go/src/net/http/server.go:1930 +0xb08
labca_1    | created by net/http.(*Server).Serve
labca_1    |    /usr/local/go/src/net/http/server.go:3034 +0x4e8
```

Besides that there seem to be no errors. I'd be glad for every hint if I maybe missed something while installing this.
The last thing I see is this:

<img width="1141" alt="image" src="https://user-images.githubusercontent.com/83862724/164934805-7357cf6d-6109-493e-9ca9-cf33ebf0abc5.png">

**Comment by hakwerk** [comment]
Hmm, the first issue is that the timezone seems to be in an unrecognized format. Could you please show the output of:
```
cat /etc/timezone
```

I haven't seen that mysql error before. Could you please show the output of
```
docker-compose ps -a
```
I wonder if the boulder application is running normally or if it is crashing and thereby perhaps taking up some mysql connections...

**Comment by schlaggi** [comment]
This is my timezone:
```
sysadmin@labCA:~$ cat /etc/timezone
Europe/Berlin
```

And this is the other output:
```
root@labCA:/home/labca/gopath/src/github.com/letsencrypt/boulder# docker-compose ps -a
      Name                     Command               State                                   Ports
-----------------------------------------------------------------------------------------------------------------------------------
boulder_bmysql_1    docker-entrypoint.sh mysql ...   Up      3306/tcp
boulder_boulder_1   labca/entrypoint.sh              Up      4001/tcp, 4002/tcp, 4003/tcp
boulder_labca_1     ./setup.sh                       Up      3000/tcp
boulder_nginx_1     /docker-entrypoint.sh ngin ...   Up      0.0.0.0:443->443/tcp,:::443->443/tcp, 0.0.0.0:80->80/tcp,:::80->80/tcp

```

**Comment by hakwerk** [comment]
It looks like the commands to determine the memory size in the "log-stats" part of the commander script give unexpected output. I'm using the "free" command to get the memory and I expect something like this as output:
```
root@testpki:/home/labca/labca# free -b --si | grep 'Mem:'
Mem:      1023885312   465862656   136855552      131072   421167104   408395776
```
But on your system that apparently gives no output. I need to improve the error handling there, but for now you could edit the /home/labca/labca/commander script and put dummy values instead of lines 140/141 like this:
```
    #total=$(free -b --si | grep 'Mem:' | perl -p0e 's/.*?\s+(\d+)\s+.*/$1/')
    total=2
    #avail=$(free -b --si | grep 'Mem:' | perl -p0e 's/.*\s+(\d+)$/$1/')
    avail=1
```
No need to restart anything, but the web GUI now should work, it will only show dummy memory data on the dashboard.

**Comment by schlaggi** [comment]
Oh I see! You grep for 'Mem:' but as my system is in german it says 'Speicher:' an that leads to no output.
Thank you for your help!

```
root@labCA:/home/sysadmin# free -b --si
              gesamt       benutzt     frei      gemns.  Puffer/Cache verfügbar
Speicher: 1023578112   667328512   126824448      487424   229425152   207847424
Swap:     1022357504   215482368   806875136
```

**Comment by hakwerk** [comment]
This should have been fixed in the v22.05 release by the way
